src/v/pandaproxy/schema_registry/json.cc

```diff
@@ -1197,7 +1223,7 @@ bool is_not_combinator_superset(
         // for not combinator, we want to check if the "not" newer subschema is
         // less strict than the older subschema, because this means that newer
         // validated less data than older
-        return is_superset(newer_it->value, older_it->value);
+        return is_superset(ctx, newer_it->value, older_it->value);
```

Comment:
need to swap the order of contexts , like newer and older are subschema are swapped.
pseudo code:
```suggestion
        return is_superset({ctx.newer, ctx.older}, newer_it->value, older_it->value);
```        

otherwise `newer` will access the context for older as `older` will access the context for newer

Reply:
Good spot!